sys/include/xtimer/tick_conversion.h

```diff
@@ -37,97 +78,142 @@ extern "C" {
+    return us; /* no-op */
 }
 
-static inline uint32_t _xtimer_ticks_from_usec(uint32_t usec) {
-    return usec; /* no-op */
+static inline uint32_t _xtimer_usec_from_ticks(uint32_t ticks)
+{
+    debug_print_ticks(XTIMER_HZ, "=", (uint64_t)ticks, "ticks", (uint64_t)ticks, "us");
+    return ticks; /* no-op */
 }
 
-static inline uint64_t _xtimer_ticks_from_usec64(uint64_t usec) {
-    return usec; /* no-op */
+static inline uint64_t _xtimer_usec_from_ticks64(uint64_t ticks)
+{
+    debug_print_ticks(XTIMER_HZ, "=", ticks, "ticks", ticks, "us");
+    return ticks; /* no-op */
 }
-
 #elif XTIMER_HZ == (32768ul)
 /* This is a common frequency for RTC crystals. We use the fact that the
  * greatest common divisor between 32768 and 1000000 is 64, so instead of
  * multiplying by the fraction (32768 / 1000000), we will instead use
  * (512 / 15625), which reduces the truncation caused by the integer widths */
-static inline uint32_t _xtimer_ticks_from_usec(uint32_t usec) {
-    return div_u32_by_15625div512(usec);
+static inline uint32_t _xtimer_ticks_from_usec(uint32_t us)
+{
+    debug_print_ticks(XTIMER_HZ, "<", (uint64_t)us, "us",
+                      (uint64_t)div_u64_by_15625div512(us), "ticks");
+    return div_u32_by_15625div512(us);
 }
 
-static inline uint64_t _xtimer_ticks_from_usec64(uint64_t usec) {
-    return div_u64_by_15625div512(usec);
+static inline uint64_t _xtimer_ticks_from_usec64(uint64_t us)
+{
+    debug_print_ticks(XTIMER_HZ, "<", us, "us", div_u64_by_15625div512(us), "ticks");
+    return div_u64_by_15625div512(us);
 }
 
-static inline uint32_t _xtimer_usec_from_ticks(uint32_t ticks) {
-    /* return (usec * 15625) / 512; */
+static inline uint32_t _xtimer_usec_from_ticks(uint32_t ticks)
+{
+    /* return (us * 15625) / 512; */
     /* Using 64 bit multiplication to avoid truncating the top 9 bits */
-    uint64_t usec = (uint64_t)ticks * 15625ul;
-    return (usec >> 9); /* equivalent to (usec / 512) */
+    uint64_t us = (uint64_t)ticks * 15625ul;
+
+    debug_print_ticks(XTIMER_HZ, "<", (uint64_t)ticks, "ticks",
+                      (uint64_t)(us >> 9), "us");
+    return (us >> 9); /* equivalent to (us / 512) */
 }
 
-static inline uint64_t _xtimer_usec_from_ticks64(uint64_t ticks) {
-    /* return (usec * 15625) / 512; */
-    uint64_t usec = (uint64_t)ticks * 15625ul;
-    return (usec >> 9); /* equivalent to (usec / 512) */
+static inline uint64_t _xtimer_usec_from_ticks64(uint64_t ticks)
+{
+    /* return (us * 15625) / 512; */
+    uint64_t us = (uint64_t)ticks * 15625ul;
+
+    debug_print_ticks(XTIMER_HZ, "<", ticks, "ticks", (us >> 9), "us");
+    return (us >> 9); /* equivalent to (us / 512) */
 }
 
 #else
 /* No matching implementation found, try to give meaningful error messages */
 #if ((XTIMER_HZ % 15625) == 0)
-#error Unsupported hardware timer frequency (XTIMER_HZ), missing XTIMER_SHIFT in board.h? See xtimer.h documentation for more info
+#error Unsupported hardware timer frequency (XTIMER_HZ).
```

Comment:
why did you changed this error message (here and below). Now the important info is only in the comments instead of showing it to the developer while compiling.

Reply:
see https://github.com/RIOT-OS/RIOT/pull/9026#discussion_r184322833